white-paper/wp.tex

```diff
@@ -65,34 +65,177 @@
 \end{abstract}
 
 % \onehalfspace
 
 \section{Statement}
+IT industry moves from centralized to cloud architecture, where information stored in distributed copies and synchronized frequently. There is a problem to evaluate whether the chosen model is suitable before start expensive development. The solution for this problem is the Model Behavior Simulator. Simulation software is used by engineers to imitate a real-world phenomenon before manufacturing the engineered product.
 
 \section{Requirements}
 \label{sec:requirements}
 
 \subsection{Features}
 \label{sec:features}
+Start statement is a huge graph of physical network components and set of repositories, placed on nodes of this graph. System allows to simulate the following things:
+\begin{itemize}
+    \item Discrete Event: Chron jobs, run by timer with or without variable parameters.
+    \item Actor-Based: When operator can make any unexpected change in any place of the system.
+    \item System Dynamics: System can change its behavior based on state of components.
+\end{itemize}
 
+\subsection{Non-unctional Requirements}
+\label{sec:nfr}
 The most critical
 \href{https://en.wikipedia.org/wiki/Non-functional_requirement}{Non-functional requirements}
 are:
 
 \todo{add non-functional requirements}
 
 \subsection{Functional Requirements}
-\label{sec:nfr}
+\label{sec:fr}
 
 The most important \href{https://en.wikipedia.org/wiki/Functional_requirement}{functional requirements} are:
 
-\todo{add functional requirements}
+\subsubsection{Input data}
+\label{sec:input}
+
+The system potentially may have different kinds of front-ends, as no integration expected. Initial and system settings might be stored as files.
+    
+System must simulate a high-loaded network of large enterprise with active users. It is expected to have the following numbers, in terms of load, size, and speed:
+
+\begin{tabular}{ll}
+  Repositories & 2M \\
+  Active users & 100K/day \\
+  Merges & 100K/day \\
+  Fetches & 15M/day, 15K/m - peak \\
+  Push & 200K/day \\
+  Traffic - download & 200Tb/day \\
+  Traffic - update & 250Gb/day \\
+\end{tabular}
+
+An algorithm for test node behavior must be implemented as external function. This function will control over nodes and able to manipulate them in a same way as ``physical'' emulated network node.
+
+Network configuration must be provided as full graph of physical servers (vertices) and their connections (edges).
+
+Probability of any network issues be provided as settings for simulator.
+
+Frequency of all possible operations and average time for processing must be set as initial parameters.
+
+To generate graph the following settings are needed to be set:
+\begin{itemize}
+    \item Number of regions;
+    \item Number of nodes;
+    \item Number of repositories;
```

Comment:
@IlyaMoskva I think we can abstract away of repositories and just call it as node state - this simulator can be potentially used not for git repository management only. Se the init script sets up initial state for node.

Reply:
@g4s8 I meant here a number how many repositories Generator should create as entities before start Simulation. The value expected here is 1.000.000, for example. And that set of repos will be "placed" on nodes.